tools/tpm2_createprimary.c

```diff
@@ -55,7 +55,7 @@
     |TPMA_OBJECT_FIXEDTPM|TPMA_OBJECT_FIXEDPARENT \
     |TPMA_OBJECT_SENSITIVEDATAORIGIN|TPMA_OBJECT_USERWITHAUTH
 
-#define DEFAULT_PRIMARY_KEY_ALG "rsa2048:aes256"
+#define DEFAULT_PRIMARY_KEY_ALG "rsa"
```

Comment:
If you walk through https://github.com/tpm2-software/tpm2-tools/blob/master/lib/tpm2_alg_util.c#L346 This would end up being:
`rsa2048:null:aes128cfb`

Also, doesn't that spec have AES(128|256)CFB as mandatory:
https://trustedcomputinggroup.org/wp-content/uploads/TCG_PC_Client_Platform_TPM_Profile_PTP_Specification_Family_2.0_Revision_1.3v22.pdf
 at Table 2?

Reply:
Sorry I don't see the problem that it ends up in `rsa2048:null:aes128cfb`. Can you please explain? If we have a problem here, then we have it in 90% of our tests using `tpm2_createprimary` with rsa key, because they use `-G rsa`.

Regarding PTP Spec:

> In Section 3.2, Table 2, the comment for TPM_ALG_AES incorrectly includes 256-bit keys as
necessary minimum support. This revision of the specification should be interpreted to
require support for only 128-bit AES keys.

see https://trustedcomputinggroup.org/wp-content/uploads/Errata_v1.1_TCG_PC_Client_Platform_TPM_Profile_PTP_2.0_r1.03_v22.pdf, 3.1 Errata 1

Reply:
> Sorry I don't see the problem that it ends up in `rsa2048:null:aes128cfb`. Can you please explain? If we have a problem here, then we have it in 90% of our tests using `tpm2_createprimary` with rsa key, because they use `-G rsa`.

`-G rsa` isn't the problem, the problem is explained in the errata below, it defaults to 128, i thought 256 was mandatory.

This also ties into issues:
  - https://github.com/tpm2-software/tpm2-tools/issues/1206
  - i think their is more, I cant find them.

We should probably pick defaults based on what the TPM actually supports and picking the best by running getcap internally versus guessing at keysize.

So the question is, do you actually want to fix the algorithm identifiers the correct way, and choose sizes and algorithms by querying the TPM?

Reply:
I set the key size to the full string, now.

I would prefer to set default algorithms to the strongest, minimal required algorithm.
And if someone wants to use the strongest supported algorithm, one can still do this using getcap externally.

Furthermore I set default aes key size to 128 in general and updated/fixed the manpages